src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DisposableFieldsShouldBeDisposed.cs

```diff
@@ -101,6 +101,17 @@ bool HasDisposeMethod(INamedTypeSymbol namedType)
                     }
                 }
 
+                foreach (var type in namedType.GetBaseTypes())
```

Comment:
This might have a performance drawback for big classes or classes with deep inheritance chain but I don't know how to make it better.

Reply:
@Evangelink Can we look at `AllInterfaces` and look for `IDisposable` or `IAsyncDisposable`?

Reply:
We already know that the type implements one of the two interfaces (check is done at the beginning), at this point we want to find one of the method (Dipose, Dispose(bool), Close, ....) that is available from current level. The idea is to say if the base class implements the interface but is not having Dispose virtual, at the current level we can't dispose so we should not report,